pkg/api/norman/customization/aks/listers.go

```diff
@@ -178,6 +191,45 @@ func listVirtualNetworks(ctx context.Context, cap *Capabilities) ([]byte, int, e
 	return encodeOutput(networks)
 }
 
+type clustersResponseBody struct {
+	ResourceGroup string `json:"resourceGroup"`
+	ClusterName   string `json:"clusterName"`
+}
+
+func listClusters(ctx context.Context, cap *Capabilities) ([]byte, int, error) {
+	clientCluster, err := NewClusterClient(cap)
+	if err != nil {
+		return nil, http.StatusInternalServerError, err
+	}
+
+	clusterList, err := clientCluster.List(ctx)
+	if err != nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("failed to get cluster list: %v", err)
+	}
+
+	var clusters []clustersResponseBody
+
+	for clusterList.NotDone() {
+		for _, cluster := range clusterList.Values() {
+			var tmpCluster clustersResponseBody
+			if cluster.ID != nil {
+				tmpCluster.ResourceGroup = matchResourceGroup.FindStringSubmatch(to.String(cluster.ID))[1]
```

Comment:
How do you know you will find a match and not hit `index out of range` error ? I think you should check the length first.